src/notifications/settings/settings.ts

```diff
@@ -1,114 +1,129 @@
+import { DEFAULT_ENABLED_TOPIC_SETTINGS } from '@/notifications/settings/constants';
 import {
-  DEFAULT_ENABLED_GLOBAL_TOPIC_SETTINGS,
-  NOTIFICATIONS_DEFAULT_CHAIN_ID,
-  WALLET_TOPICS_STORAGE_KEY,
-} from '@/notifications/settings/constants';
-import {
+  GlobalNotificationTopics,
   GlobalNotificationTopicType,
-  WalletNotificationRelationshipType,
   WalletNotificationTopicType,
   WalletNotificationSettings,
 } from '@/notifications/settings/types';
 import {
+  getAllGlobalNotificationSettingsFromStorage,
   getAllWalletNotificationSettingsFromStorage,
-  notificationSettingsStorage,
-  setAllGlobalNotificationSettingsToStorage,
+  setAllWalletNotificationSettingsToStorage,
 } from '@/notifications/settings/storage';
-import {
-  subscribeToGlobalNotificationTopic,
-  subscribeWalletToNotificationTopic,
-  unsubscribeFromAllGlobalNotificationTopics,
-  unsubscribeFromGlobalNotificationTopic,
-  unsubscribeWalletFromAllNotificationTopics,
-  unsubscribeWalletFromNotificationTopic,
-} from '@/notifications/settings/firebase';
+import { trackChangedGlobalNotificationSettings } from '@/notifications/analytics';
 
-export const removeGlobalNotificationSettings = (): Promise<void> => {
-  return unsubscribeFromAllGlobalNotificationTopics().then(() =>
-    setAllGlobalNotificationSettingsToStorage(DEFAULT_ENABLED_GLOBAL_TOPIC_SETTINGS)
-  );
-};
+import { publishWalletSettings } from '@/notifications/settings/firebase';
 
 /**
  1. Reads notification settings for all wallets from storage.
  2. Matches settings for the wallet with the given address.
  3. Excludes that wallet from the array and saves the new array.
- 4. Unsubscribes the wallet from all notification topics on Firebase.
+ 4. Updates the notification subscription
  */
-export const removeNotificationSettingsForWallet = (address: string): Promise<void> => {
-  const allSettings = getAllWalletNotificationSettingsFromStorage();
-  const settingsForWallet = allSettings.find((wallet: WalletNotificationSettings) => wallet.address === address);
+export const removeNotificationSettingsForWallet = async (address: string): Promise<void> => {
+  const walletSettings = getAllWalletNotificationSettingsFromStorage();
+  const globalSettings = getAllGlobalNotificationSettingsFromStorage();
+  const settingsForWallet = walletSettings.find((wallet: WalletNotificationSettings) => wallet.address === address);
```

Comment:
do we need to worry about case? maybe slap a `toLowerCase()` on both of these?

Reply:
there are other instances of this check but I didn't want to leave a comment on all of them.